tests/assembly/cmse.rs

```diff
@@ -9,15 +12,88 @@ pub trait Sized {}
 #[lang = "copy"]
 pub trait Copy {}
 
-// CHECK-LABEL: __acle_se_entry_point
-// CHECK: bxns
+// CHECK-LABEL: __acle_se_entry_point:
+// CHECK-NEXT: entry_point:
+//
+// Write return argument (two registers since 64bit integer)
+// CHECK: movs r0, #0
+// CHECK: movs r1, #0
+//
+// If we are using hard-float:
+// * Check if the float registers were touched (bit 3 in CONTROL)
+// hard: mrs     [[REG:r[0-9]+]], control
+// hard: tst.w   [[REG]], #8
+// hard: beq     [[LABEL:[\.a-zA-Z0-9_]+]]
+//
+// * If touched clear all float registers (d0..=d7)
+// hard: vmov    d0,
```

Comment:
Problem: err wait, I think the RHS operand is significant in this case since we want to check if the register is *cleared*. Did this previously use a register which was set to `0` or something to clear? If that is the case should we match the zero register like `[[ZERO_REG:r[0-9]+]]`, and check `vmov d0, [[ZERO_REG]]`?

Reply:
mmh what the registers are cleared with is not super relevant, for example in some cases the address of the function pointer was used. It just needs to be something an attacker can not gain information from. I'm confident that if values are written to the registers they are something that achieves this.

Except for if LLVM is behaving in an evil way, I think this is fine.

Reply:
lol, yeah, then that's fine with me 👍